app/Authentication/AuthServiceProvider.php

```diff
@@ -42,6 +43,13 @@ public function register(): void
 
         CurrentUser::configureGodUserIds($container->get('config')['user_permissions']['allow_all']);
 
+        $container->addShared(CheckTypeOfOfferMiddleware::class, function () use ($container): CheckTypeOfOfferMiddleware {
```

Comment:
This new class `CheckTypeOfOfferMiddleware` that is now part of the container also needs to be exposed with `getProvidedServiceNames`.

I also think a better location is `PsrRouterServiceProvider` or `OfferServiceProvider` because this new middleware is not related to authentication.